### PR TITLE
Enable GitHub Actions and package for macOS

### PR DESCRIPTION
**Pull request by grischard**
This re-creates the Travis test file in Github Actions.

Travis tests are currently borken because they haven't been updated for jUnit 5. These Github actions have partially been updated - maybe that's a good opportunity to switch :)

This builds upon and supercedes https://github.com/openstreetmap/josm/pull/59 (to make it easier to work on this), and there's also discussion at https://josm.openstreetmap.de/ticket/19370 . It's also related to https://josm.openstreetmap.de/ticket/18319#comment:29 and originally, when the yak shaving is done, is intended to get JOSM builds automatically signed and notarized for macOS with the Fossgis developer certificate.

CC @thomersch

### CODE REVIEW

**Comment by grischard** [comment]
All right, I think I'm pretty much happy with it. The failing tests seem to be _really_ failing :).

Improvements over Travis:

- Runs on more platforms
- CI updated to run jUnit 5 tests correctly
- Uploads all ant logs as artefacts

Future work after this PR is merged:

- Even more platforms - maybe macOS and Windows can work with headless=false?
- Build binaries on releases / nightly
- Add the Fossgis key as a github secret (Thomas or I will ideally be given access), and build a signed/notarized JOSM.app for macOS

**Comment by grischard** [comment]
Since the ultimate goal of this is to package and build for macOS, I've added those steps.

File: .github/workflows/ant.yml [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+        uses: actions/upload-artifact@v2
+        with:
+          name: Ant reports for JOSM ${{ env.josm_revision }} on java ${{ matrix.java }} on ${{ matrix.os }} with headless=${{ matrix.headless }}
+          path: test/report/*.txt
+      - name: Build and package for macOS
+        if: ${{ runner.os == 'macos' && always() }}
+        env:
+          CERT_MACOS_P12: ${{ secrets.CERT_MACOS_P12 }}
+          CERT_MACOS_PW: ${{ secrets.CERT_MACOS_PW }}
+          APPLE_ID_PW: ${{ secrets.APPLE_ID_PW }}
```
Comment (simon04) [inline .github/workflows/ant.yml:85]: Presumably, those secrets have to be added to the openstreetmap/josm repository?

Comment (thomersch) [reply .github/workflows/ant.yml:85]: Correct. I have them and can either send them to you or set them up myself, if given access to the repo.

Comment (grischard) [reply .github/workflows/ant.yml:85]: JOSM.icns has been optimised - unnecessary sizes removed, and the remaining png compressed.

**Comment by grischard** [comment]
I've also emailed Dirk about this to use this for the release builds, but he hasn't replied yet. He's probably busy - he hasn't been too active on JOSM lately.

**Comment by stoecker** [comment]
> I've also emailed Dirk about this to use this for the release builds, but he hasn't replied yet. He's probably busy - he hasn't been too active on JOSM lately.

Got no mail from you.

**Comment by simon04** [comment]
Thank you, @grischard and @thomersch, I've applied your patch via https://josm.openstreetmap.de/changeset/16776/josm

For disabling Travis CI, I've opened a separate ticket – https://josm.openstreetmap.de/ticket/19541

**Comment by grischard** [comment]
> > I've also emailed Dirk about this to use this for the release builds, but he hasn't replied yet. He's probably busy - he hasn't been too active on JOSM lately.
> 
> Got no mail from you.

Damn it. I sent it to openstreetmap at yourdomain dot de on the 11th. Any better address I should use? You can email me at guillaume openstreetmap Luxembourg, that way we'll be able to contact each other.

**Comment by stoecker** [comment]
> Damn it. I sent it to openstreetmap at yourdomain dot de on the 11th. Any better address I should use?

If yourdomain is dstoecker.de then that's fine. Got no mail to this address on 11th except for the stuff from JOSM server. I'd say simply retry.

No, there is no better address. Please remember time of sending (best with seconds). That makes checking server log much easier in case of trouble. I really get lots of mail on my server.

**Comment by grischard** [comment]
Odd. I've resent it just now.

**Comment by simon04** [comment]
@stoecker, @grischard, please resolve the outstanding discussion for release builds via https://josm.openstreetmap.de/ticket/19370

This PR has been applied, big thanks to everyone involved!

**Pull request closed without merging**